bin/citrea/tests/mempool/mod.rs

```diff
@@ -455,3 +457,84 @@ async fn test_same_nonce_tx_replacement() {
 
     seq_task.abort();
 }
+
+/// Transactions with a high gas limit should be accounted for by using
+/// their actual cumulative gas consumption to prevent them from reserving
+/// whole blocks on their own.
+#[tokio::test]
+async fn test_gas_limit_too_high() {
+    citrea::initialize_logging();
+
+    let db_dir = tempdir_with_children(&["DA", "sequencer"]);
+    let da_db_dir = db_dir.path().join("DA").to_path_buf();
+    let sequencer_db_dir = db_dir.path().join("sequencer").to_path_buf();
+
+    let (seq_port_tx, seq_port_rx) = tokio::sync::oneshot::channel();
+
+    let target_gas_limit = 15_000_000;
+    let transfer_gas_limit = 650_000;
```

Comment:
Based on what i have seen in the test, almost 23 txs included within the 15M gas limit. Each transaction consumed:
```
Tx cumulative 43615
Tx cumulative 117196
Tx cumulative 415811
Tx cumulative 436811
Tx cumulative 457811
Tx cumulative 478811
Tx cumulative 499811
Tx cumulative 520811
Tx cumulative 541811
Tx cumulative 562811
Tx cumulative 583811
Tx cumulative 604811
Tx cumulative 625811
Tx cumulative 646811
Tx cumulative 667811
Tx cumulative 688811
Tx cumulative 709811
Tx cumulative 730811
Tx cumulative 751811
Tx cumulative 772811
Tx cumulative 793811
Tx cumulative 814811
Tx cumulative 835811
Tx cumulative 856811
Tx cumulative 877811
Tx cumulative 898811
```

Which i saw on average is 650K (How come it's not 21K?)

Reply:
btw, this is from the values returned by `tx.cumulative_gas_used()`

https://github.com/chainwayxyz/citrea/pull/639/files#diff-2039c1f53f29d88ff08ce7de70335902799a60dec1fcf87ba0f54ec87a292745R366